framework/auth/core.py

```diff
@@ -667,6 +667,7 @@ def add_unclaimed_record(self, node, referrer, given_name, email=None):
             'email': clean_email,
         }
         self.unclaimed_records[project_id] = record
+        self.save()
```

Comment:
I can switch that up if you want. I just figured magic might be better than having to remember to do it. It's an easy change.